### DISCUSSION TITLE
App Services functionality to enable communication between apps and extensions

### DISCUSSION THREAD

**Discussion by FireCubeStudios** [CommunityToolkit/Labs-Windows#455] [Proposals]
Currently the Labs App Services library supports simplifying communication between a full trust uwp app and it's desktop extension. However another major use of App Services are allowing communications between Apps and also allowing communication with a uwp app and it's AppExtensions.

Having the Toolkit AppServices library support simplifying these scenarios will allow more developers to enhance their apps.

Support for handling progress, cancelation, complex types and app service contacts are a must. Additional things would need to be considered for cross app communication like knowing which app is being communicated to. I'm not too sure about how App Service works with AppExtensions so I don't know what to consider for that scenario.

**Comment by michael-hawker** [comment]
FYI @Sergio0694
